File: test_encode.js
function getScripByteArray(c, a) {
    if (c.charCodeAt[c.length - 1] == "&") {
        c = c.substring(0, c.length - 1)
    }
    let scripArray = c.split("&");
    let scripsCount = scripArray.length;
    let dataLen = 0;
    for (let index = 0; index < scripsCount; index++) {
        scripArray[index] = a + "|" + scripArray[index];
        dataLen += scripArray[index].length + 1
    }
    let bytes = new Uint8Array(dataLen + 2);
    let pos = 0;
    bytes[pos++] = ((scripsCount >> 8) & 255);
    bytes[pos++] = (scripsCount & 255);
    for (let index = 0; index < scripsCount; index++) {
        let currScrip = scripArray[index];
        let scripLen = currScrip.length;
        bytes[pos++] = (scripLen & 255);
        for (let strIndex = 0; strIndex < scripLen; strIndex++) {
            bytes[pos++] = currScrip.charCodeAt(strIndex)
        }
    }
    return bytes
}
let res = getScripByteArray("nse_cm|11536&nse_fo|51386", "sf");
console.log(Array.from(res));

File: test_encode.rs
fn build_scrip_byte_array(scrips: &str) -> Vec<u8> {
    let scrip_list: Vec<String> = scrips
        .split('&')
        .filter(|s| !s.is_empty())
        .map(|s| format!("{}|{}", "sf", s))
        .collect();

    let scrips_count = scrip_list.len();
    let data_len: usize = scrip_list.iter().map(|s| s.len() + 1).sum::<usize>() + 2;
    let mut bytes = Vec::with_capacity(data_len);

    bytes.push(((scrips_count >> 8) & 0xFF) as u8);
    bytes.push((scrips_count & 0xFF) as u8);

    for scrip in &scrip_list {
        bytes.push((scrip.len() & 0xFF) as u8);
        bytes.extend_from_slice(scrip.as_bytes());
    }
    bytes
}
fn main() {
    println!("{:?}", build_scrip_byte_array("nse_cm|11536&nse_fo|51386"));
}
